app/controllers/application_controller/ci_processing.rb

```diff
@@ -792,8 +804,19 @@ def discover
       end
       @in_a_form = true
       drop_breadcrumb(:name => "#{title} Discovery", :url => "/host/discover")
+      @discover_type_selected = params[:discover_type_selected]
 
-      if request.parameters[:controller] == "ems_cloud" || params[:discover_type_ipmi].to_s == "1"
+      if request.parameters[:controller] == "ems_cloud" && params[:discover_type_selected] == ExtManagementSystem.ems_cloud_discovery_types.first
```

Comment:
you are right model should have returned back a hash with key/value pair, i was concerned because what if the order of the array being returned is changed then it will display incorrect fields on the screen. i still think we should just check for params[:discover_type_selected] == 'azure' instead, same on line 853 below.